### DISCUSSION TITLE
Automated realtime language translations

### DISCUSSION THREAD

**Discussion by gyurmak** [EDCD/EDDI#2485] [Needs thought]
Hi There!

Is there any possibility to translate the text that will be spoken by TTS - BEFORE TTS speaks. For example all text should be translated by google translator as it comes from Journal from english to any language, and after, the translated text will be imported to EDDI
-or english text imported to EDDI and the translations made in EDDI's code - and after spoken by TTS.???
In this way we can hear the translated text in any language. (of course you may install other TTS language to fits).

Is there any solution for that?

Thanx,
George

**Comment by Tkael** [comment]
Perhaps  but with that approach we'd have far less control over the translation. Why not use the localization process which has already been established?

https://github.com/EDCD/EDDI/blob/Release/4.0.2/docs/Localization_for_translators.md

**Reply by gyurmak** [reply]
Dear Tkael,

THX for the info, I just checked it, but it is nor clear to me yet how can I install these translations to EDDI? How it will be working in realtime? Is there any tutorial for that???
THX

**Reply by Tkael** [reply]
You wouldn't install them but you would submit translations and we would include them in the next build we release. As long as a translation exists and you have selected the language in EDDI, EDDI will use that translation in place of the English.

**Reply by gyurmak** [reply]
Dear Tkael,

Ok, I understood. I tried to change lang to hungarian in EDDI, but I experienced no changes at said TTS speaking (tested on jump between systems, still english words are spoken about system names, planets surfaces, scanned results, etc.) maybe because of the hungarian translate is only 71% done?

THX for answers
george, HUngary

**Reply by Tkael** [reply]
Yes, and perhaps also because there is currently no translation of the default scripts into Hungarian.
